File: parfumerie-frontend/src/app/services/fournisseur.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { FournisseurDTO } from '../models/fournisseur.model';

@Injectable({
  providedIn: 'root'
})
export class FournisseurService {

  private baseUrl = 'http://localhost:8080/fournisseur';
  
    constructor(private http: HttpClient) { }
  
    creer(employe: FournisseurDTO): Observable<FournisseurDTO> {
      return this.http.post<FournisseurDTO>(`${this.baseUrl}/create`, employe);
    }
  
    modifier(id: number, employe: FournisseurDTO): Observable<FournisseurDTO> {
      return this.http.put<FournisseurDTO>(`${this.baseUrl}/update/${id}`, employe);
    }
  
    consulterById(id: number): Observable<FournisseurDTO> {
      return this.http.get<FournisseurDTO>(`${this.baseUrl}/consult/${id}`);
    }
  
    consulter(): Observable<FournisseurDTO[]> {
      return this.http.get<FournisseurDTO[]>(`${this.baseUrl}/consult`);
    }
  
    supprimer(id: number): Observable<void> {
      return this.http.delete<void>(`${this.baseUrl}/delete/${id}`);
    }
}

File: parfumerie-frontend/parfumerie-backend/src/main/java/com/parfumerie/gestion_parfumerie/services/FournisseurService.java
package com.parfumerie.gestion_parfumerie.services;

import java.util.List;

import com.parfumerie.gestion_parfumerie.dto.FournisseurDTO;


public interface FournisseurService {

	FournisseurDTO creer(FournisseurDTO fournisseurDTO);
	FournisseurDTO modifier(Long id, FournisseurDTO fournisseurDTO);
	FournisseurDTO consulterById(Long id);
	List<FournisseurDTO> consulter();
	void supprimer(Long id);
}
